src/main/java/net/mcreator/element/parts/gui/Slot.java

```diff
@@ -32,7 +33,8 @@ public abstract class Slot extends GUIComponent {
 
 	public Color color;
 	public int id;
-	public boolean disableStackInteraction;
+	public LogicProcedure disablePickupCondition;

```

Comment:
I would rename those to disablePickup and disablePlacement only without condition, as this is now not necessarily condition but can also be a value. Check ItemExtension#fuelPower where we used similar naming convention where we assumed the field is number, not a procedure number provider